pkg/controller/machineautoscaler/machinetarget.go

```diff
@@ -81,3 +121,68 @@ func (mt *MachineTarget) GetLimits() (min, max int, err error) {
 
 	return min, max, nil
 }
+
+// SetOwner sets the target's owner annotation to the given object.  It returns
+// a boolean indicating whether the owner annotation changed, and an error,
+// which will be ErrTargetAlreadyOwned if the target is already owned.
+func (mt *MachineTarget) SetOwner(owner metav1.Object) (bool, error) {
+	annotations := mt.GetAnnotations()
+
+	if annotations == nil {
+		annotations = make(map[string]string)
+	}
+
+	ownerRef := types.NamespacedName{
+		Namespace: owner.GetNamespace(),
+		Name:      owner.GetName(),
+	}
+
+	if a, ok := annotations[MachineTargetOwnerAnnotation]; ok {
+		if a != ownerRef.String() {
+			return false, ErrTargetAlreadyOwned
+		}
+
+		return false, nil
+	}
+
+	annotations[MachineTargetOwnerAnnotation] = ownerRef.String()
+	mt.SetAnnotations(annotations)
+
+	return true, nil
+}
+
+// RemoveOwner removes the owner annotation from the target.
+func (mt *MachineTarget) RemoveOwner() {
+	annotations := mt.GetAnnotations()
+
+	delete(annotations, MachineTargetOwnerAnnotation)
+
+	mt.SetAnnotations(annotations)
```

Comment:
At face value the call to `Setannotations` looks redundant. We Get, then Set without appearing to change anything - curious as to why?

Reply:
We delete the owner annotation. So we get the annotations map, remove the owner annotation, then set the resulting map with no owner back on the target.